packages/remix-dev/compiler/plugins/cssModulesPlugin.ts

```diff
@@ -0,0 +1,387 @@
+  filePath,
+}: {
+  config: RemixConfig;
+  filePath: string;
+}): Promise<CssModuleFileContents> {
+  // TODO: Caching as implemented not working as expected in dev mode. Bypass
+  // for now.
+  return await processCss({ config, filePath });
+
+  //   let file = path.resolve(config.appDirectory, filePath);
+  //   let hash = await getFileHash(file);
+  //   // Use an on-disk cache to speed up dev server boot.
+  //   let processedCssPromise = (async function () {
+  //     let key = file + ".cssmodule";
+  //     let cached: (CssModuleFileContents & { hash: string }) | null = null;
+  //     try {
+  //       cached = await cache.getJson(config.cacheDirectory, key);
+  //     } catch (error) {
+  //       // Ignore cache read errors.
+  //     }
+  //     if (!cached || cached.hash !== hash) {
+  //       let processed = await processCss({ config, filePath });
+  //       cached = { hash, ...processed };
+  //       try {
+  //         await cache.putJson(config.cacheDirectory, key, cached);
+  //       } catch (error) {
+  //         // Ignore cache put errors.
+  //       }
+  //     }
+  //     return cached;
+  //   })();
+  //   return processedCssPromise;
+}
+
+export async function processCss({
+  config,
+  filePath,
+}: {
+  config: RemixConfig;
+  filePath: string;
+}): Promise<CssModuleFileContents> {
+  let classPrefix =
+    path.basename(filePath, path.extname(filePath)).replace(/\./g, "-") + "__";
+  let source = await fse.readFile(filePath);
+
+  let res = parcelTransform({
+    filename: path.relative(config.appDirectory, filePath),
+    code: source,
+    cssModules: true,
+    minify: process.env.NODE_ENV === "production",
+    analyzeDependencies: true, // TODO: Maybe?
+    sourceMap: true,
+    drafts: { nesting: true },
+  });
+
+  let json: CssModuleClassMap = {};
+  let cssModulesContent = decoder.decode(res.code);
+  let parcelExports = res.exports || {};
+
+  // sort() to keep order consistent in different builds
+  for (let originClass of Object.keys(parcelExports).sort()) {
+    let props = parcelExports[originClass];
+    let patchedClass = props.name;
+    let prefixedClassName = getPrefixedClassName(classPrefix, patchedClass);
+    json[originClass] = props.composes.length
+      ? getComposedClassNames(classPrefix, prefixedClassName, props.composes)
+      : prefixedClassName;
+    cssModulesContent = cssModulesContent.replace(
+      new RegExp(`\\.${patchedClass}`, "g"),
+      "." + prefixedClassName
+    );
+  }
+
+  let cssWithSourceMap = cssModulesContent;
+  if (res.map) {
+    // TODO: Sourcemaps aren't working as expected because we are inlining the
+    // map at the end of each module. We need to merge them into a single
+    // inline sourcemap at the end of the build. We can probably use something
+    // like https://www.npmjs.com/package/merge-source-maps
+    // cssWithSourceMap += `\n/*# sourceMappingURL=data:application/json;base64,${res.map.toString(
+    //   "base64"
+    // )} */`;
```

Comment:
You could try out the Parcel source map library for this potentially: https://github.com/parcel-bundler/source-map